src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,299 @@
+	 */
+	public static function register( $module_identifier, $src, $dependencies = array(), $version = false ) {
+		if ( ! isset( self::$registered[ $module_identifier ] ) ) {
+			$deps = array();
+			foreach ( $dependencies as $dependency ) {
+				if ( isset( $dependency['id'] ) ) {
+					$deps[] = array(
+						'id'   => $dependency['id'],
+						'type' => isset( $dependency['type'] ) && 'dynamic' === $dependency['type'] ? 'dynamic' : 'static',
+					);
+				} elseif ( is_string( $dependency ) ) {
+					$deps[] = array(
+						'id'   => $dependency,
+						'type' => 'static',
+					);
+				}
+			}
+
+			self::$registered[ $module_identifier ] = array(
+				'src'          => $src,
+				'version'      => $version,
+				'enqueued'     => in_array( $module_identifier, self::$enqueued_before_registered, true ),
+				'dependencies' => $deps,
+			);
+		}
+	}
+
+	/**
+	 * Marks the module to be enqueued in the page.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $module_identifier The identifier of the module.
+	 */
+	public static function enqueue( $module_identifier ) {
+		if ( isset( self::$registered[ $module_identifier ] ) ) {
+			self::$registered[ $module_identifier ]['enqueued'] = true;
+		} elseif ( ! in_array( $module_identifier, self::$enqueued_before_registered, true ) ) {
+			self::$enqueued_before_registered[] = $module_identifier;
+		}
+	}
+
+	/**
+	 * Unmarks the module so it is no longer enqueued in the page.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string $module_identifier The identifier of the module.
+	 */
+	public static function dequeue( $module_identifier ) {
+		if ( isset( self::$registered[ $module_identifier ] ) ) {
+			self::$registered[ $module_identifier ]['enqueued'] = false;
+		}
+		$key = array_search( $module_identifier, self::$enqueued_before_registered, true );
+		if ( false !== $key ) {
+			array_splice( self::$enqueued_before_registered, $key, 1 );
+		}
+	}
+
+	/**
+	 * Returns the import map array.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array Array with an `imports` key mapping to an array of module identifiers and their respective URLs, including the version query.
+	 */
+	public static function get_import_map() {
+		$imports = array();
+		foreach ( self::get_dependencies( array_keys( self::get_enqueued() ) ) as $module_identifier => $module ) {
+			$imports[ $module_identifier ] = $module['src'] . self::get_version_query_string( $module['version'] );
+		}
+		return array( 'imports' => $imports );
+	}
+
+	/**
+	 * Prints the import map using a script tag with a type="importmap" attribute.
+	 *
+	 * @since 6.5.0
+	 */
+	public static function print_import_map() {
+		$import_map = self::get_import_map();
+		if ( ! empty( $import_map['imports'] ) ) {
+			echo '<script type="importmap">' . wp_json_encode( self::get_import_map(), JSON_HEX_TAG | JSON_HEX_AMP ) . '</script>';
```

Comment:
Can we use `wp_print_inline_script_tag` here?

Reply:
Changed in https://github.com/WordPress/wordpress-develop/pull/5818/commits/bbae17c72567a67a09487faffa73fe160218f904.

It adds a CDATA wrapper, but I guess that's ok, isn't it?

```html
<script type="importmap">
  /* <![CDATA[ */
  { "imports": { "dep": "\/dep.js?ver=6.5-alpha-56966-src" } }
  /* ]]> */
</script>
```